app/src/main/java/org/simple/clinic/scheduleappointment/ScheduleAppointmentSheetController.kt

```diff
@@ -77,126 +61,18 @@ class ScheduleAppointmentSheetController @AssistedInject constructor(
-    ui.enableDecrementButton(areEarlierPotentialAppointmentsAvailable)
-  }
-
-  private fun scheduleOnExactDate(
-      events: Observable<UiEvent>
-  ): Observable<UiChange> {
-    return events
-        .ofType<AppointmentCalendarDateSelected>()
-        .map { it.selectedDate }
-        .map(::generatePotentialAppointmentDate)
-        .doOnNext { latestAppointmentScheduled = it }
-        .map { appointmentDate ->
-          { ui: Ui ->
-            ui.updateScheduledAppointment(appointmentDate.scheduledFor, appointmentDate.timeToAppointment)
-            enableIncrements(ui)
-            enableDecrements(ui)
-          }
-        }
-  }
-
-  private fun scheduleOnPreviousConfiguredAppointmentDate(
-      events: Observable<UiEvent>
-  ): Observable<UiChange> {
-    return events
-        .ofType<AppointmentDateDecremented>()
-        .map { previousConfiguredAppointmentDate(latestAppointmentScheduled) }
-        .doOnNext { latestAppointmentScheduled = it }
-        .map { appointmentDate ->
-          { ui: Ui ->
-            ui.updateScheduledAppointment(appointmentDate.scheduledFor, appointmentDate.timeToAppointment)
-            enableIncrements(ui)
-            enableDecrements(ui)
-          }
-        }
-  }
-
-  private fun scheduleOnNextConfiguredAppointmentDate(
-      events: Observable<UiEvent>
-  ): Observable<UiChange> {
-    return events
-        .ofType<AppointmentDateIncremented>()
-        .map { nextConfiguredAppointmentDate(latestAppointmentScheduled) }
-        .doOnNext { latestAppointmentScheduled = it }
-        .map { appointmentDate ->
-          { ui: Ui ->
-            ui.updateScheduledAppointment(appointmentDate.scheduledFor, appointmentDate.timeToAppointment)
-            enableIncrements(ui)
-            enableDecrements(ui)
-          }
-        }
-  }
-
-  private fun scheduleDefaultAppointmentDateForSheetCreates(events: Observable<UiEvent>): Observable<UiChange> {
-    val protocolUuid = Observable.fromCallable { currentFacility.get().protocolUuid!! }
-    val protocolStream = protocolUuid.switchMap(protocolRepository::protocol)
-
-    val configTimeToAppointment = Observable.fromCallable { config.defaultTimeToAppointment }
-    val protocolTimeToAppointment = protocolStream.map { Days(it.followUpDays) }
-
-    val timeToAppointments = Observable
-        .concatArrayEager(configTimeToAppointment, protocolTimeToAppointment)
-        .subscribeOn(schedulers.io())
-
-    return Observables
-        .combineLatest(
-            events.ofType<ScreenCreated>(),
-            timeToAppointments
-        ) { _, timeToAppointment -> timeToAppointment }
-        .map(::generatePotentialAppointmentDate)
-        .doOnNext { latestAppointmentScheduled = it }
-        .map { appointmentDate ->
-          { ui: Ui ->
-            ui.updateScheduledAppointment(appointmentDate.scheduledFor, appointmentDate.timeToAppointment)
-            enableIncrements(ui)
-            enableDecrements(ui)
-          }
-        }
-  }
```

Comment:
I don't understand the point of breaking these into four streams if we were going to delete this code in the next commit?
Can we squash these commits?

Reply:
Yes, I broke these up first because I planned to migrate them one-by-one, but then I realized later on that's not possible.

I can squash them.

Reply:
Done.